SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -1,271 +1,214 @@
-						"%lld"),
-				   EntityId);
-			CompletedRequests.Add(EntityId);
-			continue;
-		}
-
-		const SpatialGDK::ComponentPresence* ComponentPresenceComponent =
-			StaticComponentView->GetComponentData<SpatialGDK::ComponentPresence>(EntityId);
-		if (ComponentPresenceComponent == nullptr)
-		{
-			// This happens if the ComponentPresence component is removed in the same tick as a request is queued, but the request was not
-			// removed from the queue - shouldn't happen.
-			UE_LOG(
-				LogSpatialLoadBalanceEnforcer, Error,
-				TEXT("Cannot process entity as ComponentPresence component has been removed since the request was queued. EntityId: %lld"),
-				EntityId);
-			CompletedRequests.Add(EntityId);
-			continue;
-		}
-
-		if (AuthorityIntentComponent->VirtualWorkerId == SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning,
-				   TEXT("Entity with invalid virtual worker ID assignment will not be processed. EntityId: %lld. This should not happen - "
-						"investigate if you see this warning."),
-				   EntityId);
-			CompletedRequests.Add(EntityId);
-			continue;
-		}
-
-		check(VirtualWorkerTranslator != nullptr);
-		const PhysicalWorkerName* DestinationWorkerId =
-			VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
-		if (DestinationWorkerId == nullptr)
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Error,
-				   TEXT("This worker is not assigned a virtual worker. This shouldn't happen! Worker: %s"), *WorkerId);
-			continue;
-		}
+		ComponentIds.AddUnique(RequiredComponentId);
+	}
 
-		if (!StaticComponentView->HasAuthority(EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Log,
-				   TEXT("Failed to update the EntityACL to match the authority intent; this worker lost authority over the EntityACL since "
-						"the request was queued."
-						" Source worker ID: %s. Entity ID %lld. Desination worker ID: %s."),
-				   *WorkerId, EntityId, **DestinationWorkerId);
-			CompletedRequests.Add(EntityId);
-			continue;
-		}
+	// Get the client worker ID net-owning this Actor from the NetOwningClientWorker.
+	const PhysicalWorkerName PossessingClientId =
+		NetOwningClientWorkerComponent.WorkerId.IsSet() ? NetOwningClientWorkerComponent.WorkerId.GetValue() : FString();
 
-		TArray<Worker_ComponentId> ComponentIds;
+	const FString& WriteWorkerId = FString::Printf(TEXT("workerId:%s"), **DestinationWorkerId);
 
-		EntityAcl* Acl = StaticComponentView->GetComponentData<EntityAcl>(EntityId);
-		Acl->ComponentWriteAcl.GetKeys(ComponentIds);
+	const WorkerAttributeSet OwningServerWorkerAttributeSet = { WriteWorkerId };
 
-		// Ensure that every component ID in ComponentPresence is set in the write ACL.
-		for (const auto& RequiredComponentId : ComponentPresenceComponent->ComponentList)
+	for (const Worker_ComponentId& ComponentId : ComponentIds)
```

Comment:
Nit: this probably does not need to be a const ref

Reply:
kept the constness as I think that still means something to the reader, but ref is gone.